google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/readrows/LargeReadRowsResumptionStrategy.java

```diff
@@ -0,0 +1,169 @@
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.stub.readrows;
+
+import com.google.api.core.InternalApi;
+import com.google.api.gax.retrying.StreamResumptionStrategy;
+import com.google.api.gax.rpc.ApiException;
+import com.google.bigtable.v2.ReadRowsRequest;
+import com.google.bigtable.v2.ReadRowsRequest.Builder;
+import com.google.bigtable.v2.RowSet;
+import com.google.cloud.bigtable.data.v2.internal.RowSetUtil;
+import com.google.cloud.bigtable.data.v2.models.RowAdapter;
+import com.google.cloud.bigtable.data.v2.stub.BigtableStreamResumptionStrategy;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ByteString;
+import java.util.Base64;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * An implementation of a {@link StreamResumptionStrategy} for merged rows. This class tracks -
+ *
+ * <ul>
+ *   <li>row key for the last row that was read successfully
+ *   <li>row key for large-row that couldn't be read
+ *   <li>list of all row keys for large-rows
+ * </ul>
+ *
+ * Upon retry this class builds a request to omit the large rows & retry from the last row key that
+ * was successfully read.
+ *
+ * <p>This class is considered an internal implementation detail and not meant to be used by
+ * applications.
+ */
+@InternalApi
+public class LargeReadRowsResumptionStrategy<RowT>
+    extends BigtableStreamResumptionStrategy<ReadRowsRequest, RowT> {
+  private final RowAdapter<RowT> rowAdapter;
+  private ByteString lastSuccessKey = ByteString.EMPTY;
+  // Number of rows processed excluding Marker row.
+  private long numProcessed;
+  private ByteString largeRowKey = ByteString.EMPTY;
+  private final AtomicInteger largeRowsCount = new AtomicInteger(0);
+
+  private ReadRowsRequest originalRequest;
+
+  private ReadRowsRequest lastModifiedRequestOnTimeOfError;
+
+  private RowSet previousFailedRequestRowset = null;
+
+  public LargeReadRowsResumptionStrategy(RowAdapter<RowT> rowAdapter) {
+    this.rowAdapter = rowAdapter;
+  }
+
+  @Override
+  public boolean canResume() {
+    return true;
+  }
+
+  @Override
+  public StreamResumptionStrategy<ReadRowsRequest, RowT> createNew() {
+    return new LargeReadRowsResumptionStrategy<>(rowAdapter);
+  }
+
+  @Override
+  public RowT processResponse(RowT response) {
+    // Last key can come from both the last processed row key and a synthetic row marker. The
+    // synthetic row marker is emitted when the server has read a lot of data that was filtered out.
+    // The row marker can be used to trim the start of the scan, but does not contribute to the row
+    // limit.
+    lastSuccessKey = rowAdapter.getKey(response);
+
+    if (!rowAdapter.isScanMarkerRow(response)) {
+      // Only real rows count towards the rows limit.
+      numProcessed++;
+    }
+    return response;
+  }
+
+  public Throwable processError(Throwable throwable) {
+    String rowKeyExtracted = extractLargeRowKey(throwable);
+    if (rowKeyExtracted != null) {
```

Comment:
Let's log the large rows here. You can add a logger to this class with 

```
private static final Logger LOGGER = Logger.getLogger(LargeReadRowsResumptionStratigy.class.getName());
```

and here add 

```
LOGGER.wraning("skipping large row " + rowKeyExtracted);
```